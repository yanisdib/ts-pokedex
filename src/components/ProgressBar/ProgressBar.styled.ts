import styled from 'styled-components';


interface ProgressBarProps {
    progressValue: number;
}


export const StyledProgressBar = styled.div`
    display: flex;
    width: 80%;
    height: 18px;
    border: 2px solid #E0E0E0;
    border-radius: 20px;

    &::before {
        content: "";
        width: ${({ 
            progressValue
        }: ProgressBarProps) => progressValue}%;
        left: 0;
        background-color: red;
        transition: width 0.2s ease-in;
        border-radius: 20px;
    }
`;

export const ProgressIndicator = styled.span`
    font-family: var(--info-font);
    font-weight: 500;
`;