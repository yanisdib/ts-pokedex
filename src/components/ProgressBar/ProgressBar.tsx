import { ProgressIndicator, StyledProgressBar } from './ProgressBar.styled';


interface Props {
    progressValue: number;
}


function ProgressBar({ progressValue }: Props): JSX.Element {
    return (
        <>
            <StyledProgressBar progressValue={progressValue} />
            <ProgressIndicator>
                {progressValue < 100 ? (
                    `${progressValue.toFixed(0)}%`
                ) : (
                    'COMPLETE'
                )}
            </ProgressIndicator>
        </>
    );
}


export default ProgressBar;