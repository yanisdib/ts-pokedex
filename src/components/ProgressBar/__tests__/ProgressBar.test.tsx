import { render } from '@testing-library/react';

import ProgressBar from '../ProgressBar';


describe('ProgressBar component', () => {
    test('should render without crashing', () => {
        const { container } = render(<ProgressBar progressValue={0} />);

        expect(container).toMatchSnapshot();
    });

    test('should display a progression of 30%', () => {
        const { getByTestId, getByText } = render(
            <ProgressBar
                progressValue={30}
            />
        );

        expect(getByText('30%')).toBeInTheDocument();
    });
});