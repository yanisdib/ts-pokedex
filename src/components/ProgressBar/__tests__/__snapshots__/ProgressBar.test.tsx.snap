// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component should render without crashing 1`] = `
<div>
  <div
    class="sc-bczRLJ gxJOsm"
  />
  <span
    class="sc-gsnTZi kGfZrx"
  >
    0%
  </span>
</div>
`;
